### PR TITLE
tkt-78672: Add support for getting and setting ACLs through middleware

### PR DESCRIPTION
**Pull request by anodos325**
setacl and getacl:
- include ability to manipulate ACLs with simplified permset and flagset
simple permset: FULL_CONTROL, MODIFY, READ, TRAVERSE, NOPERMS
simple flagset: INHERIT, NOINHERIT

setacl:
- if setacl is called with a path and the option "strip=true", then
the ACL already set on the file will be converted to a trivial ACL (one in
which the ACL permset and flagset is fully expressed in the mode).
- middleware job because 'winacl -a clone' can be very longrunning.
Current record is about 3-4 hours for about 60TB of data.

Additional fix:
- some variables in winacl needed to be initialized to zero.

to do:
- FTS_XDEV flag in winacl doesn't prevent us from setting the ACL directly on a sub-
dataset (but it does protect the contents). I need to make this a little more
intelligent.

- Calculation of ACL to be inherited in winacl also needs to be a little more
precise. (remove non-inheriting ACEs and properly handle 'inherit-only' and
'no-propagate inherit' bits).

### CODE REVIEW

File: src/middlewared/middlewared/plugins/filesystem.py [hunk 1]
```diff
@@ -227,6 +229,152 @@ def statfs(self, path):
...
+                }
+                if ace['tag'] == 'everyone@' and ace['perms'] == 'NOPERMS':
+                    continue
+                simple_acl.append(ace)
+
+            return simple_acl
+
+    @accepts(
+        Str('path'),
+        List('dacl', default=[]),
```
Comment (william-gr) [inline src/middlewared/middlewared/plugins/filesystem.py]: I think we should have something like this in List dacl:

`items=[Dict('daclentry', Str('tag'), ...]`

File: src/middlewared/middlewared/utils/acltools.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
+# -*- coding=utf-8 -*-
+import enum
+from bsd import acl
+
+"""
+These are simplified forms of permissions sets based
+on NTFS basic permissions.
+"""
```
Comment (william-gr) [inline src/middlewared/middlewared/utils/acltools.py]: Do you imagine us ever using this outside filesystem plugin? If not embedding this in the plugin makes sense to me.

Comment (anodos325) [reply src/middlewared/middlewared/utils/acltools.py]: That's a good point. Cleaning this up.

File: src/middlewared/middlewared/plugins/filesystem.py (hunk 1 above)
Comment (anodos325) [reply src/middlewared/middlewared/plugins/filesystem.py]: Done.

**Pull request merged**